pkg/util/client_util.go

```diff
@@ -19,6 +20,9 @@ func NewClientWithCachedReader(ctx context.Context, config *rest.Config, scheme
 	}
 	go ccache.Start(ctx)
 	ccache.WaitForCacheSync(ctx)
+	if !ccache.WaitForCacheSync(ctx) {
+		return nil, nil, fmt.Errorf("failed to sync cache")
+	}
```

Comment:
```suggestion
	if !ccache.WaitForCacheSync(ctx) {
		return nil, nil, fmt.Errorf("failed to sync cache")
	}
```

Shouldn't we call the synchronizing cache only once?

Reply:
We don't put a global cacheReader to include all potential CR schemes; instead, every plugin registers its own schema and hence instantiate its own cacheReader and call its own WaitForCacheSync.

@tenzen-y Do you see some side effect here?

Reply:
No, I do not observe side effects in the actual cluster. I was just wondering why we should check the cache sync twice.

Reply:
> I was just wondering why we should check the cache sync twice.

What do you mean by "twice" here? (to ensure I fully understand your concern)

Reply:
Ah sry... my bad

Reply:
Both line 22 and 23 execute waiting cache sync: https://github.com/Huang-Wei/scheduler-plugins/blob/65d77ef02af42b23b760c4c25b71cdab6ce5f4af/pkg/util/client_util.go#L22-L23

Reply:
there is a typo...

Reply:
https://github.com/kubernetes-sigs/scheduler-plugins/pull/899 will fix this.